brack-parser/src/angle.rs

```diff
@@ -1,27 +1,25 @@
 use anyhow::Result;
 use brack_tokenizer::tokens::{mock_location, Token};
 
-use crate::error::ParserError;
+use crate::error::{DocumentError, ParseTerminationError, ParserError};
 use crate::{ast::new_angle, parser::Parser, surrounded, utils::consume_by_kind};
 
 // "<" ident (expr ("," expr)*)? ">"
 pub fn parse(tokens: &Vec<Token>) -> Result<Parser, ParserError> {
     let (mut consumed, mut new_tokens) =
         consume_by_kind(&tokens, Token::AngleBracketOpen(mock_location()));
     if !consumed {
-        return Err(ParserError::new_parse_termination_error(
-            "Angle Brackets is not opened.".to_string(),
-            tokens.first().unwrap().clone(),
-        ));
+        return Err(ParseTerminationError::AngleNotOpened(
+            new_tokens.first().unwrap().get_location(),
+        )
+        .into());
     }
     let mut result = new_angle();
 
     match surrounded::parse(&new_tokens) {
         Ok((asts, tokens)) => {
             for ast in asts {
-                result.add(ast).map_err(|e| {
-                    ParserError::new_document_error(e.to_string(), tokens[0].clone())
-                })?;
+                result.add(ast).unwrap();
```

Comment:
Why do you need `unwrap()` for `result.add(ast)`?

Reply:
Looking at the [implementation](https://github.com/brack-lang/brack/blob/66c8f60960ccea50e1319c627abf10696d2f9ae3/brack-sdk-rs/src/ast.rs#L70-L86) of `ast.add()`, it is designed not to return an error. However, the reason `ast.add()` still returns a `Result` type is to notify any improper operations when plugins manipulate the AST in the future. Therefore, such an error cannot occur during parsing, and I judged it appropriate to use `unwrap`.

Reply:
I see. It will certainly never pass invalid ast to add, but what I mean is that there are enough `result.add(ast)` for you to ignore result values.

Reply:
I'm sorry that what I said before is wrong. The Rust compiler warns against unhandling the return value. I suggest you write the following.

```rs
result.add(ast).expect("This should never fail because it occurs when an invalid AST is inserted by a plugin macro, so it is safe to `parse`.")
```

Reply:
I think this explanation should be included as a comment, so I added the comment and committed the changes.